src/style-spec/function/evaluation_context.js

```diff
@@ -0,0 +1,170 @@
+'use strict';
+
+const parseColor = require('../util/parse_color');
+const interpolate = require('../util/interpolate');
+
+class RuntimeError extends Error {
+    constructor(message) {
+        super(message);
+        this.name = 'ExpressionEvaluationError';
+        this.message = message;
+    }
+
+    toJSON() {
+        return `${this.name}: ${this.message}`;
+    }
+}
+
+class Color {
+    constructor(input) {
+        if (Array.isArray(input)) {
+            this.value = input;
+        } else {
+            this.value = parseColor(input);
+            if (typeof this.value === 'undefined') throw new RuntimeError(`Could not parse color from value '${input}'`);
+        }
+    }
+}
+
+module.exports = () => ({
+    get: function (obj, key) {
+        if (!obj) throw new RuntimeError(`Cannot get property ${key} from null object`);
+        if (!(key in obj)) throw new RuntimeError(`Property ${key} not found in object with keys: [${Object.keys(obj)}]`);
+        return obj[key];
+    },
+
+    typeOf: function (x) {
+        if (Array.isArray(x)) return 'Vector<Value>';
```

Comment:
It's not quite the only place it's exposed publicly: type checking error messages will need to refer to types by name, and probably some runtime error messages will, too.  That said, I'm also with you in that I've been uneasy/unsatisfied with having `Vector<Value>` as a possible output of `typeof` (because of both the generics syntax and the `Value` alias).

Would it be confusing for `[typeof, [get, 'foo']]` to equal `Array`, but then to have `[number,
 [at, [get, 'foo'], 0]]` to produce a runtime error message like `Expected Vector<number> but properties.foo was an object instead`?